### DISCUSSION TITLE
Plugins: Hook for new call?

### DISCUSSION THREAD

**Discussion by rzvc** [vimeo/psalm#7976] [Q&A]
Hey,

If someone has insight into this, I would really appreciate the help. I'm struggling with adding a hook for AfterNewAnalysis - I don't believe there is one.

I want to do that so I can help with the inference of the type parameter for that particular instance. Is this possible?

Thanks.

Edit: There were two questions here, only one remains.

**Comment by orklah** [comment]
There is no AfterNew hook indeed, however, we do have an AfterExpressionAnalysis which should include "New" statements. Can't you use that?

**Reply by rzvc** [reply]
I don't know, I'll give it a try. Currently I switched to the factory model, so I can make use of AfterMethodCallAnalysisInterface.

**Reply by orklah** [reply]
Both should work yeah
